test/framework/src/main/java/org/elasticsearch/entitlement/bootstrap/TestEntitlementsRule.java

```diff
@@ -0,0 +1,357 @@
+import org.elasticsearch.core.PathUtils;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.entitlement.initialization.EntitlementInitialization;
+import org.elasticsearch.entitlement.runtime.policy.PathLookup;
+import org.elasticsearch.entitlement.runtime.policy.PathLookup.BaseDir;
+import org.elasticsearch.entitlement.runtime.policy.Policy;
+import org.elasticsearch.entitlement.runtime.policy.PolicyManager;
+import org.elasticsearch.entitlement.runtime.policy.PolicyParser;
+import org.elasticsearch.entitlement.runtime.policy.TestPathLookup;
+import org.elasticsearch.entitlement.runtime.policy.TestPolicyManager;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.plugins.PluginDescriptor;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+
+import static java.util.stream.Collectors.toCollection;
+import static java.util.stream.Collectors.toSet;
+import static org.elasticsearch.entitlement.runtime.policy.PathLookup.BaseDir.TEMP;
+import static org.elasticsearch.env.Environment.PATH_DATA_SETTING;
+import static org.elasticsearch.env.Environment.PATH_HOME_SETTING;
+import static org.elasticsearch.env.Environment.PATH_REPO_SETTING;
+import static org.elasticsearch.env.Environment.PATH_SHARED_DATA_SETTING;
+
+public class TestEntitlementsRule implements TestRule {
+    private static final Logger logger = LogManager.getLogger(TestEntitlementsRule.class);
+
+    private static final Map<BaseDir, Collection<Path>> BASE_DIR_PATHS = new ConcurrentHashMap<>();
+    private static final TestPolicyManager POLICY_MANAGER;
+    private static final AtomicBoolean active = new AtomicBoolean(false);
+
+    static {
+        PathLookup pathLookup = new TestPathLookup(BASE_DIR_PATHS);
+        try {
+            if (isEnabledForTests()) {
+                POLICY_MANAGER = createPolicyManager(pathLookup);
+                loadAgent(POLICY_MANAGER, pathLookup);
+            } else {
+                POLICY_MANAGER = null;
+            }
+        } catch (IOException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    public static void initialize(Path tempDir) throws IOException {
+        if (POLICY_MANAGER != null) {
```

Comment:
the tempDir is only ever java.io.tmpdir right? So couldn't this be known statically and not need this initialize method?